.github/workflows/test.yml

```diff
@@ -64,79 +53,25 @@ jobs:
-          git switch ${{ github.event.pull_request.head.ref }}
-          git branch --show-current
-
-      - name: Commit `test/bench` changes
-        if: github.actor != 'dependabot[bot]' && runner.os == 'Linux' && runner.arch == 'X64' && github.event_name == 'pull_request' && matrix.build_type == 'release'
-        uses: EndBug/add-and-commit@v9
-        with:
-          default_author: github_actions
-          author_name: Cycle and memory benchmark updater
-          github_token: ${{ secrets.NIV_UPDATER_TOKEN }}
-          committer_name: GitHub Actions
-          committer_email: actions@github.com
-          message: "Updating `test/bench` numbers"
-          commit: -- test/bench/ok
-
-      - name: Calculate performance delta
-        if: github.actor != 'dependabot[bot]' && runner.os == 'Linux' && runner.arch == 'X64' && github.event_name == 'pull_request' && matrix.build_type == 'release'
-        run: |
-          from="$(git merge-base origin/${{ github.base_ref }} HEAD)"
-          to="${{ github.event.pull_request.head.sha }}"
-          echo "Comparing changes from $from to $to"
-          nix build -f perf-delta.nix -o perf-delta.txt \
-            --argstr ref HEAD \
-            --argstr from "$from" \
-            --argstr to "$to"
-
-      - name: Read performance delta
-        if: github.actor != 'dependabot[bot]' && runner.os == 'Linux' && runner.arch == 'X64' && github.event_name == 'pull_request' && matrix.build_type == 'release'
-        id: perf
-        uses: juliangruber/read-file-action@v1
-        with:
-          path: ./perf-delta.txt
-
-      - name: Find performance comment
-        if: github.actor != 'dependabot[bot]' && runner.os == 'Linux' && runner.arch == 'X64' && github.event_name == 'pull_request' && matrix.build_type == 'release'
-        uses: peter-evans/find-comment@v3
-        id: fc
-        with:
-          issue-number: ${{ github.event.pull_request.number }}
-          comment-author: 'github-actions[bot]'
-          body-includes: '<!-- perf comment -->'
-
-      # Forks can't add comments so this job does not run on forks, see
-      # motoko#2864.
-      - name: Create or update performance comment
-        if: github.actor != 'dependabot[bot]' && runner.os == 'Linux' && runner.arch == 'X64' && github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository && matrix.build_type == 'release'
-        uses: peter-evans/create-or-update-comment@v4
+          os: ${{ matrix.os }}
+          test-target: ${{ matrix.build_type }}-systems-go
+          test-name: ${{ matrix.os }}-${{ matrix.build_type }}-tests
+          cachix-auth-token: ${{ secrets.CACHIX_AUTH_TOKEN }}
+
+      - name: Run Performance Tests
+        if: github.actor != 'dependabot[bot]' && github.event_name == 'pull_request' && runner.os == 'Linux' && runner.arch == 'X64' && matrix.build_type == 'release'
```

Comment:
I wonder if we can make these conditions centrally and just use a bool field here...

Reply:
I'm looking into it, I think there's a way.

Reply:
hmm, actually that doesn't help much because some of the variables used there e.g. `runner.*`, or `matrix.*` are only populated for the step afaik.